### PR TITLE
Error bar UI tests converted to use fake adapter

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

 ErrorBar UI tests converted to use fake adapter instead of real adapters

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/2845227

# How Tested

Tested locally. Local run time comparison: 

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/tests/browser/chat/app/FakeChatAdapterArgs.ts [hunk 1]
```diff
@@ -58,4 +58,10 @@ export type FakeChatAdapterArgs = {
    * Determines if chat composite will be using a custom data model
    */
   customDataModelEnabled?: boolean;
+  /*
+   * Record of rest errors to throw when functions of ChatThreadClient are called
+   */
+  chatThreadClientFunctionErrors?: Partial<Record<keyof ChatThreadClient, ChatThreadRestError>>;
```
Comment (prprabhu-ms) [inline packages/react-composites/tests/browser/chat/app/FakeChatAdapterArgs.ts:64]: Naming nit: `chatThreadClientMethodErrors`. An interface / class has methods, not functions.

File: packages/react-composites/tests/browser/chat/fake-adapter/ErrorBar.test.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    );
+    await waitForChatCompositeToLoad(page);
+    await stubMessageTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('error-bar-wrong-thread-id.png');
+
+    await sendMessage(page, TEST_MESSAGE);
+    await waitForSendMessageFailure(page);
+    await stubMessageTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('error-bar-send-message-with-wrong-thread-id.png');
+    // test resend button in contextual menu
```
Comment (prprabhu-ms) [inline packages/react-composites/tests/browser/chat/fake-adapter/ErrorBar.test.ts:31]: Tangent: You didn't add this check in this PR, so I agree you should carry it over.

But we should split off tests for resend button from tests for showing the correct error bar. Especially with the speed up we'll get with the fake services, we can push to make our tests be more specific.

File: packages/react-composites/tests/browser/chat/fake-adapter/ErrorBar.test.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+const TEST_MESSAGE = 'No, sir, this will not do.';
+
+test.describe('ErrorBar is shown correctly', async () => {
+  test('with wrong thread ID', async ({ page, serverUrl }) => {
+    await page.goto(
+      buildUrlForChatAppUsingFakeAdapter(serverUrl, {
+        ...DEFAULT_FAKE_CHAT_ADAPTER_ARGS,
+        chatThreadClientFunctionErrors: {
+          getProperties: { message: 'Could not get properties', statusCode: 400 },
```
Comment (prprabhu-ms) [inline packages/react-composites/tests/browser/chat/fake-adapter/ErrorBar.test.ts:17]: Did you check these return codes by running the real test?

**Review by prprabhu-ms — APPROVED** [review]

File: packages/react-composites/tests/browser/chat/fake-adapter/ErrorBar.test.ts (hunk 3 above)
Comment (mgamis-msft) [reply packages/react-composites/tests/browser/chat/fake-adapter/ErrorBar.test.ts:17]: Yes I did.

**Review by dmceachernmsft — APPROVED** [review]

**Pull request merged**